tests/mocked.cpp

```diff
@@ -0,0 +1,164 @@
+//  Copyright 2018 U.C. Berkeley RISE Lab
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "mocked.hpp"
+
+std::string MockZmqUtil::message_to_string(const zmq::message_t& message) {
+  return std::string(static_cast<const char*>(message.data()), message.size());
+}
+zmq::message_t MockZmqUtil::string_to_message(const std::string& s) {
```

Comment:
For these implementations that are shared, can't we just put them in the superclass?

Reply:
fixed